pageserver/src/tenant/ephemeral_file.rs

```diff
@@ -21,18 +21,42 @@ use crate::tenant::storage_layer::inmemory_layer::vectored_dio_read::File;
 use crate::virtual_file::owned_buffers_io::io_buf_aligned::IoBufAlignedMut;
 use crate::virtual_file::owned_buffers_io::slice::SliceMutExt;
 use crate::virtual_file::owned_buffers_io::write::{Buffer, FlushTaskError};
-use crate::virtual_file::{self, IoBufferMut, VirtualFile, owned_buffers_io};
+use crate::virtual_file::{self, IoBufferMut, TempVirtualFile, VirtualFile, owned_buffers_io};
+
+use self::owned_buffers_io::write::OwnedAsyncWriter;
 
 pub struct EphemeralFile {
     _tenant_shard_id: TenantShardId,
     _timeline_id: TimelineId,
     page_cache_file_id: page_cache::FileId,
     bytes_written: u64,
-    buffered_writer: owned_buffers_io::write::BufferedWriter<IoBufferMut, VirtualFile>,
+    // Always Some except during Drop
+    inner: Option<Inner>,
```

Comment:
I think this is outdated. It is always `Some`, so we can drop the `Option` entirely and inline `Inner`. Unless there's a missing drop handler.

Reply:
https://github.com/neondatabase/neon/pull/11549/commits/a12c23d0be2f66956838f1904b36784923aa0edc